router.go

```diff
@@ -19,60 +19,41 @@ type Router struct {
 }
 
 // ServeHTTP implements http.Handler.
-func (r Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	r.baseRouter.ServeHTTP(w, req)
 }
 
-// NewRouter returns a new Router.
-func NewRouter(
-	doc *Document,
-	handlers OperationHandlers,
-	options ...RouterOption,
-) (Router, error) {
-	// Apply argument options.
-	router := Router{}
-	for _, o := range options {
-		o(&router)
-	}
-
-	// Default options
-	if router.debugLog == nil {
-		router.debugLog = func(format string, args ...interface{}) {}
-	}
-	if router.baseRouter == nil {
-		router.baseRouter = DefaultBaseRouter()
-	}
-
-	// Apply router-wide middlewares
-	for _, mw := range router.wrappers {
-		router.baseRouter.Use(mw)
-	}
+// AddSpec adds routes from the spec to the router.
+func (r *Router) AddSpec(doc *Document, handlers OperationHandlers) error {
+	return r.applySpecRouting(doc, handlers)
+}
 
+func (r *Router) applySpecRouting(doc *Document, handlers OperationHandlers) error {
 	// Serve the specification itself if enabled.
-	if router.serveSpec != 0 {
+	if r.serveSpec != 0 {
 		var specHandler http.Handler
-		switch router.serveSpec {
+		switch r.serveSpec {
 		case SpecHandlerTypeDynamic:
 			specHandler = DynamicSpecHandler(doc.OrigSpec())
 		case SpecHandlerTypeStatic:
 			specHandler = StaticSpecHandler(doc.OrigSpec())
 		}
-		router.baseRouter.Route(http.MethodGet, doc.Spec().BasePath, specHandler)
+		r.baseRouter.Route(http.MethodGet, doc.Spec().BasePath, specHandler)
```

Comment:
Once again - it makes impossible to serve app from `/` (to have `basePath` empty).

Reply:
It is not rebased yet, sorry :3

Reply:
Rebased on your commit, done